onnxruntime/hosting/executor.cc

```diff
@@ -24,67 +24,75 @@ namespace hosting {
-    if (!status.IsOK()) {
-      LOGS(*logger, ERROR) << "GetSizeInBytesFromTensorProto() FAILED! Input name: " << input_name
-                           << " Error code: " << status.Code()
-                           << ". Error Message: " << status.ErrorMessage();
-      return GenerateProtoBufStatus(status, "GetSizeInBytesFromTensorProto() FAILED: " + status.ErrorMessage());
-    }
+  OrtCallback deleter;
+  status = onnxruntime::utils::TensorProtoToMLValue(onnxruntime::Env::Default(), nullptr, input_tensor,
+                                                    onnxruntime::MemBuffer(data.release(), cpu_tensor_length, *cpu_allocator_info),
+                                                    ml_value, deleter);
+  if (!status.IsOK()) {
+    LOGS(*logger, ERROR) << "TensorProtoToMLValue() failed. Message: " << status.ToString();
+    return GenerateProtobufStatus(status, "TensorProtoToMLValue() failed:" + status.ErrorMessage());
+  }
 
-    std::unique_ptr<char[]> data(new char[cpu_tensor_length]);
-    if (nullptr == data) {
-      LOGS(*logger, ERROR) << "Run out memory. Input name: " << input_name;
-      return protobufutil::Status(protobufutil::error::Code::RESOURCE_EXHAUSTED, "Run out of memory");
-    }
-    memset(data.get(), 0, cpu_tensor_length);
+  return protobufutil::Status::OK;
+}
+
+protobufutil::Status Executor::SetNameMLValueMap(onnxruntime::NameMLValMap& name_value_map, const onnxruntime::hosting::PredictRequest& request) {
+  auto logger = env_->GetLogger(request_id_);
+
+  OrtAllocatorInfo* cpu_allocator_info = nullptr;
+  auto ort_status = OrtCreateAllocatorInfo("Cpu", OrtArenaAllocator, 0, OrtMemTypeDefault, &cpu_allocator_info);
+  if (ort_status != nullptr || cpu_allocator_info == nullptr) {
+    LOGS(*logger, ERROR) << "OrtCreateAllocatorInfo failed";
+    return protobufutil::Status(protobufutil::error::Code::RESOURCE_EXHAUSTED, "OrtCreateAllocatorInfo() failed");
+  }
+
+  // Prepare the MLValue object
+  for (const auto& input : request.inputs()) {
+    using_raw_data = using_raw_data && input.second.has_raw_data();
 
-    // TensorProto -> MLValue
     MLValue ml_value;
-    OrtCallback deleter;
-    status = onnxruntime::utils::TensorProtoToMLValue(onnxruntime::Env::Default(), nullptr, input_tensor,
-                                                      onnxruntime::MemBuffer(data.release(), cpu_tensor_length, *cpu_allocator_info),
-                                                      ml_value, deleter);
-    if (!status.IsOK()) {
-      LOGS(*logger, ERROR) << "TensorProtoToMLValue() FAILED! Input name: " << input_name
-                           << " Error code: " << status.Code()
-                           << ". Error Message: " << status.ErrorMessage();
-      return GenerateProtoBufStatus(status, "TensorProtoToMLValue() FAILED:" + status.ErrorMessage());
+    auto status = SetMLValue(input.second, cpu_allocator_info, ml_value);
+    if (status != protobufutil::Status::OK) {
+      return status;
```

Comment:
If it is failed, it will be good to log the Input name here. It is better to have the input name as part of the error message. Then if there are more than one inputs, the user will be easy to find out which input data has problem.

Reply:
In general, the more information you provide in the response, the less work our DRI will have. :-)